### ISSUE TITLE
How to finetune on Custom Loss function

### ISSUE THREAD

**Issue by gopal86** [time-series-foundation-models/lag-llama#38]
Hi, I wanted to fine tune the model on my own dataset however with my own custom loss. Could you give an example on how to do that?
Thanks

**Comment by ashok-arjun** [comment]
Hi @gopal86. What kind of loss functions would you like to finetune it with?

**Comment by jinxzed** [comment]
Hi, I would like to train on weighted mape loss function. Kindly share an example

**Comment by ashok-arjun** [comment]
You would have to replace the last layer with an appropriate layer for the loss function, and then finetune the model.

You could try finetuning just the last layer, or the entire model.

**Comment by gopal86** [comment]
> Hi @gopal86. What kind of loss functions would you like to finetune it with?

I have certain constraints which I would like to impose. Could you give an example on how to fine-tune any custom loss function?

**Comment by ashok-arjun** [comment]
When you finetune on any other function, you will have to change the last layer to whatever is appropriate.

For instance, if you would like to perform classification, you could put a linear layer with softmax at the end after the embeddings, and then use the cross entropy loss.

If you would like to perform regression, you could put a linear layer at the end with a single output, and use the MSE loss.

When training, you could train the full model or just the last layer.

An idea: To start with, you could get the embeddings from the pretrained model, and then train a simple statistical classifier/regressor on the embeddings.

**Comment by jinxzed** [comment]
Hi,

**Comment by jinxzed** [comment]
Hi,
Can you share an example on how to get the pretrained model embeddings for
the lag-llama model.
